components/core/src/package/install.rs

```diff
@@ -231,15 +231,27 @@ impl PackageInstall {
         // In most cases, this does nothing and should only mutate
         // the paths in a windows studio where FS_ROOT_PATH will
         // be the studio root path (ie c:\hab\studios\...)
-        for path in &mut paths {
-            *path = fs::fs_rooted_path(&path, &self.fs_root_path);
-        }
+        let rooted_path = self.root_paths(&mut paths)?;
 
-        let joined = env::join_paths(paths)?.into_string()
-                                            .map_err(Error::InvalidPathString)?;
         // Only insert a PATH entry if the resulting path string is non-empty
-        if !joined.is_empty() {
-            env.insert(PATH_KEY.to_string(), joined);
+        if !rooted_path.is_empty() {
+            env.insert(PATH_KEY.to_string(), rooted_path);
+        }
+
+        // release 0.85.0 introduces the RUNTIME_ENVIRONMENT_PATHS metadata
+        // that are environment variables containing file paths that are to be
+        // rooted under FS_ROOT. For backwards compatibility the key/value
+        // pairs are duplicated in RUNTIME_ENVIRONMENT so we will remove previously
+        // stored values with the same key.
+        let environment_paths =
+            self.runtime_environment_file_map(MetaFile::RuntimeEnvironmentPaths)?;
+        for (key, value) in environment_paths.into_iter() {
+            let mut split_path: Vec<PathBuf> = env::split_paths(&value).collect();
+            let rooted_path = self.root_paths(&mut split_path)?;
+            if !rooted_path.is_empty() {
+                env.remove(&key);
+                env.insert(key, rooted_path);
```

Comment:
It looks like [`insert`](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.insert) will replace the old value so I don/t think the `remove` is necessary.